File: ML in Python & R/Part 2 - Regression/Section 4 - Simple Linear Regression/MyWork/simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Tue Mar 24 21:07:22 2020

@author: chath
"""

## Data Preprocessing

# Import libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

## Import dataset
dataset = pd.read_csv('Salary_Data.csv')

x = dataset.iloc[:, :-1].values # only remove last column
y = dataset.iloc[:, 1].values

## Split dataset to train and test
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 1/3, random_state = 0)

## Feature scaling 
"""
# There are two ways Standardisation and Normalisation
from sklearn.preprocessing import StandardScaler

sc_x = StandardScaler()
# For the train data set we need to fit and transaform
x_train = sc_x.fit_transform(x_train)
# sc_x already fitted to training set, so just need transaform
x_test = sc_x.transform(x_test)
"""

## Simple linear regression
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(x_train, y_train)

## Predict the Test set results
y_pred = regressor.predict(x_test)

## Visulalizing the trainging set results
plt.scatter(x_train, y_train, color = 'red')
plt.plot(x_train, regressor.predict(x_train), color = 'blue') # plot with x train, y predictions for x train
plt.title('Salary vs Experience (Traning set)')
plt.xlabel('Years')
plt.ylabel('Salary')
plt.show()

## Visulalizing the test set results
plt.scatter(x_test, y_test, color = 'red')
plt.plot(x_train, regressor.predict(x_train), color = 'blue') # We don't need to change this test data cuz it's already trained using train data
plt.title('Salary vs Experience (Test set)')
plt.xlabel('Years')
plt.ylabel('Salary')
plt.show()

File: ML in Python & R/Part 2 - Regression/Section 4 - Simple Linear Regression/MyWork/simple_linear_regression.R

### Data preprocessing

## Import dataset
dataset = read.csv('Salary_Data.csv')
# dataset = dataset[, 2:3]

## Split dataset to train and test

# Need a new package caTools
# install.packages('caTools')
library(caTools) #using this we can import caTools directly

# In python random_state, in R it is seed
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3) # SplitRatio is train set ratio oposite of python, This returns True - decide to goto train/ False- goto test set

training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# ## Feature scaling
# # Facot is not numeric
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])


## Fitting Simple linear regression to training data set
regressor = lm(formula = Salary ~ YearsExperience, # Salary ~ YearsExperience | means salary propotional to YearsExperience 
                data = training_set)
# In console type summary(regressor) to get a summary
# Coefficient
# *** stars mean highly statistical significant
# Pr Value (P value) - lower the value is more significant, less than 5% mean high impact

## Predict the Test set results
y_pred = predict(regressor, newdata = test_set)

## Visulalizing the trainging set results
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             color = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') + # aes use for separate x and y axises
  ggtitle('Salary vs Experience (Traning Set)') +
  xlab('Years of Experience') +
  ylab('Salary')


## Visulalizing the test set results
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             color = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') + # No need to change this from previous, regressor already trained
  ggtitle('Salary vs Experience (Test Set)') +
  xlab('Years of Experience') +
  ylab('Salary')

